tfs/writer.py

```diff
@@ -234,35 +306,115 @@ def _dtype_to_id_string(type_: type) -> str:
     Returns:
         The ID string.
     """
-    if pdtypes.is_integer_dtype(type_) or pdtypes.is_bool_dtype(type_):
+    if pdtypes.is_integer_dtype(type_):
         return "%d"
     if pdtypes.is_float_dtype(type_):
         return "%le"
     if pdtypes.is_string_dtype(type_):
         return "%s"
-    errmsg = f"Provided type '{type_}' could not be identified as either a bool, int, float or string dtype"
+    if pdtypes.is_bool_dtype(type_):
+        return "%b"
```

Comment:
We always write booleans in boolean form (true / false) and not numerical form (0 / 1).